Tetfuza/TetfuzaBackend/InputChecker.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+namespace Tetfuza
+{
+    public interface IConsole
+    {
+        public ConsoleKey ReadKey();
+        public void Clear();
+        public bool KeyAvailable { get; }
+    }
+
+    public class InputChecker
+    {
+        private IConsole _keyboard;
```

Comment:
I'm using "dependency injection" here, to allow this code to be more testable. This way I can inject a fake console that my tests control, to simulate users clicking buttons on the keyboard.